tests/test_core.py

```diff
@@ -170,6 +170,132 @@ def test_create_interpolator(simple_grid):
     assert np.allclose(interp(field), linear_value)
 
 
+def test_set_cell_volumes():
+    """
+    Test that cell volumes are set properly.
+    """
+# Test cartesian volumes
```

Comment:
Comments need to be indented. Also, instead of comments here, you should just break this up into multiple shorter tests.